src/lib.rs

```diff
@@ -390,58 +409,65 @@ pub trait Trait: system::Trait + Sized {
 }
 
 decl_storage! {
-    // A macro for the Storage trait, and its implementation, for this module.
-    // This allows for type-safe usage of the Substrate storage database, so you can
-    // keep things around between blocks.
-    trait Store for Module<T: Trait> as Example {
-        // Any storage declarations of the form:
-        //   `pub? Name get(getter_name)? [config()|config(myname)] [build(|_| {...})] : <type> (= <new_default_value>)?;`
-        // where `<type>` is either:
-        //   - `Type` (a basic value item); or
-        //   - `map KeyType => ValueType` (a map item).
-        //
-        // Note that there are two optional modifiers for the storage type declaration.
-        // - `Foo: Option<u32>`:
-        //   - `Foo::put(1); Foo::get()` returns `Some(1)`;
-        //   - `Foo::kill(); Foo::get()` returns `None`.
-        // - `Foo: u32`:
-        //   - `Foo::put(1); Foo::get()` returns `1`;
-        //   - `Foo::kill(); Foo::get()` returns `0` (u32::default()).
-        // e.g. Foo: u32;
-        // e.g. pub Bar get(bar): map T::AccountId => Vec<(BalanceOf<T>, u64)>;
-        //
-        // For basic value items, you'll get a type which implements
-        // `support::StorageValue`. For map items, you'll get a type which
-        // implements `support::StorageMap`.
-        //
-        // If they have a getter (`get(getter_name)`), then your module will come
-        // equipped with `fn getter_name() -> Type` for basic value items or
-        // `fn getter_name(key: KeyType) -> ValueType` for map items.
-        Dummy get(dummy) config(): Option<BalanceOf<T>>;
+    trait Store for Module<T: Trait> as Forum {
+
+        /// Map category identifier to corresponding category.
+        pub CategoryById get(category_by_id) config(): map CategoryId => Category<T::BlockNumber, T::Moment, T::AccountId>;
```

Comment:
All map values should return `Option`. Otherwise, on the client side, it's impossible to understand whether there is an entity by a certain id or not. If there is no entry by id, then Substrate's RPC returns a struct with default values, i.e. `0` for numbers, `false` for booleans.
```suggestion
        pub CategoryById get(category_by_id) config(): map CategoryId => Option<Category<T::BlockNumber, T::Moment, T::AccountId>>;
```

Reply:
As all the discussion back and forth showed, we need to have a broader discussion on this - and then try to finally conclude. I can then change to `Optional` at that time, the change will be small. It has caused huge pains before to change this due to really weird possible bugs in Substrate.